Yafc/Windows/ProjectPageSettingsPanel.cs

```diff
@@ -71,7 +71,7 @@ public class ProjectPageSettingsPanel : PseudoScreen {
 
                 gui.allocator = RectAllocator.LeftRow;
                 if (editingPage != null && gui.BuildRedButton("Delete page")) {
-                    if (editingPage.canDelete) {
+                    if (editingPage.contentType != typeof(Summary)) {
```

Comment:
So back then I wanted to prevent to adding specific support in a generic location. If no-one objects this is fine by me...

Reply:
For future reference, why is specific support in a generic location considered bad? Genuinely curious.

Reply:
It adds dependencies/couple between unrelated pieces of code. Now some settings panel needs to know about the Summary page. If the Summary page is changed, we might need to change this as well.

Before, a property was introduced to add a generic feature (`canDelete`) and this code only needed to support this feature, without know about the page(s) that require it.

Reply:
I guess a more proper way to do it is to put the canDelete back in the page, but change the check to what's writen here. That way settingPanel stays decoupled from editingPage, and the editingPage only needs to know about deletion, not the how. I can change that.